plone/protect/configure.zcml

```diff
@@ -42,14 +53,6 @@
       file="protect.js"
       />
 
-  <browser:page
-      name="confirm-action"
-      for="*"
-      class=".views.ConfirmView"
-      template="confirm.pt"
-      permission="zope2.View"
-      />
-
```

Comment:
I am pretty sure the confirm view should stay in this package because we have code that mentions it: https://github.com/plone/plone.protect/blob/213f1c4da8cc7ce330edcc813868bbcdc2d1eecb/plone/protect/auto.py#L304-L308

Let me think about that.

CC @plone/classicui-team

Reply:
An interesting thing would be raising a `CSRFExceptionOrNameItLikeYouPrefer` exception instead of redirecting to a view.
Then the confirm-action page would rather be the `index.html` view for that custom exception:

See: https://github.com/plone/Products.CMFPlone/blob/6e2b2f55fc251a0e68b5ba65c09de6be867a9d39/Products/CMFPlone/browser/configure.zcml#L250-L257

I have no idea if that could even work, but sounds really sane to me.
`Products.CMFPlone` will raise a normal error page,
Classic UI -> will show the "index.html"  with the form to confirm the action.
Volto -> Whatever

Reply:
@mauritsvanrees You're right, it doesn't make sense to redirect to the confirmation in plone.restapi. But I also don't want to remove the check entirely; it's useful for finding unexpected write-on-reads. I would suggest not doing the redirect if the request only has application/json in the Accept header.

@ale-rt Using exception views is a nice idea, but the check happens in the transform chain and runs on both PubBeforeCommit and PubBeforeAbort events in the publisher. We can raise an exception during PubBeforeCommit and the publisher will look up the exception view, but if we raise an exception during PubBeforeAbort it is too late. https://github.com/zopefoundation/Zope/blob/master/src/ZPublisher/WSGIPublisher.py#L233

Reply:
> @ale-rt Using exception views is a nice idea, but the check happens in the transform chain and runs on both PubBeforeCommit and PubBeforeAbort events in the publisher. We can raise an exception during PubBeforeCommit and the publisher will look up the exception view, but if we raise an exception during PubBeforeAbort it is too late. https://github.com/zopefoundation/Zope/blob/master/src/ZPublisher/WSGIPublisher.py#L233

Ouch! Bummer :)

Reply:
> Devils advocate question: If this is the case, why then move the templates to plone.app.layout?

Reading David's answer, it seems `plone.protect` is still useful for Volto/RestApi as it prevents write-on-read.  The problem is then only that the error message you get (the confirmation page) does not get through to the Volto frontend.  So for application/json requests, we could return a json error response.  And then Volto should somehow handle this.

Since on the ClassicUI side only one page template is involved, it could be fine to keep that here.  It is not a template that anyone is likely to want to customize.

Reply:
IMO, the best solution is to keep a simple template for covering whoever uses this package without classic UI, and then have a template targeted for ClassicUI in the frontend.
I am waiting for the pt to land in plone.app.layout to port there some markup improvements from a customer project:
![image](https://github.com/user-attachments/assets/70069fe6-7a5b-4871-8f35-efae9b5a5f90)
(hoping that a good designer will make it less fleshy :rofl: )